packages/eas-cli/src/credentials/ios/appstore/__tests__/capabilityIdentifiers-test.ts

```diff
@@ -0,0 +1,170 @@
+    await syncCapabilityIdentifiersForEntitlementsAsync(bundleId, {
+      'com.apple.developer.in-app-payments': ['merchant.bacon'],
+      'com.apple.security.application-groups': ['group.bacon'],
+      'com.apple.developer.icloud-container-identifiers': ['iCloud.bacon'],
+    });
+
+    // Ensure we create missing ids
+    expect(Apple.MerchantId.createAsync).toHaveBeenLastCalledWith(
+      // auth context
+      {},
+      // props
+      { identifier: 'merchant.bacon' }
+    );
+    // Ensure we create missing ids
+    expect(Apple.AppGroup.createAsync).toHaveBeenLastCalledWith(
+      // auth context
+      {},
+      // props
+      { identifier: 'group.bacon' }
+    );
+    // Ensure we create missing ids
+    expect(Apple.CloudContainer.createAsync).toHaveBeenLastCalledWith(
+      // auth context
+      {},
+      // props
+      { identifier: 'iCloud.bacon' }
+    );
+
+    expect(bundleId.updateBundleIdCapabilityAsync).toBeCalledWith([
+      {
+        capabilityType: 'APP_GROUPS',
+        option: 'ON',
+        relationships: {
+          appGroups: ['XXX-group-2'],
+        },
+      },
+      {
+        capabilityType: 'APPLE_PAY',
+        option: 'ON',
+        relationships: {
+          merchantIds: ['XXX-merch-2'],
+        },
+      },
+      {
+        capabilityType: 'ICLOUD',
+        option: 'ON',
+        relationships: {
+          cloudContainers: ['XXX-icloud-2'],
+        },
+      },
+    ]);
+  });
+
+  it(`removes local duplicates`, async () => {
+    const Apple = require('@expo/apple-utils');
+    mockCapabilities(Apple);
+    const bundleId = {
+      context: {},
+      updateBundleIdCapabilityAsync: jest.fn(),
+      id: 'XXX',
+    } as any;
+
+    await syncCapabilityIdentifiersForEntitlementsAsync(bundleId, {
+      'com.apple.developer.in-app-payments': ['merchant.expo', 'merchant.expo'],
+    });
+
+    // Only called once because we remove local duplicates to minimize network requests.
+    expect(Apple.MerchantId.getAsync).toBeCalledTimes(1);
+  });
+
+  it(`creates missing capability identifiers`, async () => {
+    const Apple = require('@expo/apple-utils');
+    mockCapabilities(Apple);
+    const bundleId = {
+      context: {},
+      updateBundleIdCapabilityAsync: jest.fn(),
+      id: 'XXX',
+    } as any;
+
+    await syncCapabilityIdentifiersForEntitlementsAsync(bundleId, {
+      'com.apple.developer.in-app-payments': ['merchant.expo'],
+    });
+
+    expect(Apple.MerchantId.createAsync).not.toBeCalled();
+  });
+
+  it(`throws when creating a missing capability that is reserved`, async () => {
+    const Apple = require('@expo/apple-utils');
+    mockCapabilities(Apple);
+    Apple.MerchantId.createAsync = jest.fn((ctx: any, { identifier }: { identifier: string }) => {
+      // e2e test with: merchant.expodemo
+      throw new Error(
+        `There is a problem with the request entity - A Merchant ID with Identifier '${identifier}' is not available. Please enter a different string.`
```

Comment:
(i know this is a test but the code for this doesn't live in this repo so commenting here)

you might want to avoid "Please" - iirc this was something that you personally don't like in CLIs 😄 

maybe something like this:

> There is a problem with the request entity - the Merchant ID with Identifier '${identifier}' is already created and belongs to a different Apple Developer account. Provide a different value.

Reply:
This is a generated error message from the Apple API, I think it's best to just extend it so users can more easily search to find related errors in other tools that wrap the Apple ASC API.